domain/model/service/service.go

```diff
@@ -542,3 +589,59 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// getWatchActivatedModelsMapper returns a mapper function that filters change events to
+// include only those associated with activated models.
+func getWatchActivatedModelsMapper(st State) func(ctx context.Context, db database.TxnRunner,
+	changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+
+	return func(ctx context.Context, db database.TxnRunner,
+		changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+		modelUUIDs := make([]string, len(changes))
+
+		for i, change := range changes {
+			modelUUID := change.Changed()
```

Comment:
This could just become......

```
modelUUIDs := make([]coremodel.UUID, 0, len(changes)
for _, change := range changes {
    modelUUIDS = append(modelUUIDS, coremodel.UUID(change.Change()))
}
```

In my opinion this is a lot clearer and we just get the job done.

Reply:
Sure will change it from string to coremodel.UUID if it's clearer. I'm using direct assignment instead of append based on your previous suggestion to preallocate memory. I thought it would have slightly lesser overhead too since we are not checking bounds like append(), but happy to refactor to append() if I misunderstood you.

https://github.com/juju/juju/pull/19071#discussion_r1984377310